### ISSUE TITLE
Cannot handle path when working from WSL path in Windows

### ISSUE THREAD

**Issue by alecxvs** [andy-5/wslgit#71]
The insider build for 1903 has a new functionality that allows browsing files on WSL from explorer. The paths look like `\\wsl$\Ubuntu\home\alecxvs\Git\myproject`.

Running wslgit from this path gives this error:
`thread 'main' panicked at 'Cannot handle path "\\\\wsl$\\Ubuntu\\home\\alecxvs\\Git\\myproject"', src\libcore\option.rs:1008:5`

**Comment by andy-5** [comment]
Yes, it would be nice to support these new paths. Right now, `wslgit` assumes that Windows paths start with a drive letter (to translate them into a proper unix path).

Adding support for these paths should be possible, e.g. by detecting and removing the prefix.

**Comment by roonie007** [comment]
+1

**Comment by chadrien** [comment]
> Adding support for these paths should be possible, e.g. by detecting and removing the prefix.

@andy-5 why don't you use `wslpath` for things like that? E.g. `wsl.exe wslpath -u "\\\\wsl$\\Ubuntu\\home\\alecxvs\\Git\\myproject"` would return `/home/alecxvs/Git/myproject`. Would it make your life easier? :)

**Comment by andy-5** [comment]
Thanks for the suggestion (and good to know that it supports these paths). There is already some discussion on `wslpath` in #12. Unfortunately, it does not solve all problems. The main reasons it is not used are:
- it does not fully support manually mounted drives
- executing a separate process is slow
- nobody has done the work to implemented support for it

**Comment by rennex** [comment]
It should be super simple to convert WSL2 UNC paths: just add/remove the `\\wsl$\Distroname` and flip the backslashes. No need to resolve mount paths in that case, though you probably need to pass the distro name to the Linux side. However, I'm not sure if there's some special consideration needed for running executables when the current directory is a UNC path. At least .bat scripts have problems with those.

Ideally, this would detect the UNC paths and also work with DrvFs or 9p mounts, so it'd work whether the git repo resides on the Windows side or the Linux side.

**Comment by andy-5** [comment]
I just released v0.9.0, which uses `wslpath` to translate paths (and should handle UNC paths). Does the new version resolve this issue?

**Comment by Stanzilla** [comment]
Not quite, I have a repo at

`\\wsl$\Ubuntu\home\stan\projects\personal\WeakAuras2`

and when I go to it via CMD (which is T:\home\stan\projects\personal\WeakAuras2) and type `git status` it does not detect it as a git repo.

**Comment by Stanzilla** [comment]
@carlolars any idea for this maybe?

**Comment by Stanzilla** [comment]
but T IS the WSL mount, how would I mount it inside of WSL?

**Comment by carlolars** [comment]
> 
> 
> but T IS the WSL mount, how would I mount it inside of WSL?

You can't.

**Comment by Stanzilla** [comment]
Well what do I do then? :D
